src/afl-fuzz-one.c

```diff
@@ -530,6 +569,33 @@ u8 fuzz_one_original(afl_state_t *afl) {
 
   memcpy(out_buf, in_buf, len);
 
+  /**************
+   * FRAMESHIFT *
+   **************/
+
+  if (unlikely(afl->afl_env.afl_frameshift_enabled)) {
+
+    if (unlikely(afl->queue_cur->fs_status == 0)) {
+
+      /* FrameShift has not run on this input. */
+
+      /* Check frameshift overhead budget before running analysis.
+         Use total runtime since start as the reference. */
+      u64 total_runtime_ms =
+          afl->prev_run_time + get_cur_time() - afl->start_time;
+      double max_overhead = afl->afl_env.afl_frameshift_max_overhead;
+      u64    allowed_ms = (u64)((double)total_runtime_ms * max_overhead);
+
+      if (afl->fs_stats.total_time_ms < allowed_ms) { frameshift_stage(afl); }
```

Comment:
you could also allow just favorite entries - or maybe both, time budget and favorite .

Reply:
to be the best queue entry for at least one edge ("best" is execution speed and size)